From c587b9abf8f155643a30aaca7fe2a0aa1b8e6216 Mon Sep 17 00:00:00 2001
From: Dave Gerlach <d-gerlach@ti.com>
Date: Thu, 5 Jan 2017 15:32:22 -0600
Subject: [PATCH 055/205] soc: ti: wkup_m3_ipc: Fix misassigned struct device
 pointer

This commit comes from:
  git://git.ti.com/processor-sdk/processor-sdk-linux.git

The firmware loading callback for the volt scaling firmware expects the
the second argument to be a pointer to a struct wkup_m3_ipc, however the
function also attempts to cast it to be used directly as a struct device
pointer. This incorrect pointer is only used for dev_err messages so if
a failure occurs it will result in a crash.

Instead let's use the struct device pointer inside the struct
wkup_m3_ipc pointer that is actually passed so that we have a valid
dev pointer.

Fixes: 1be271922084 ("soc: ti: wkup_m3_ipc: Add support for i2c voltage
scaling")
Signed-off-by: Dave Gerlach <d-gerlach@ti.com>
Acked-by: Suman Anna <s-anna@ti.com>

(cherry picked from commit 19694ee14597e42681a9b8ed74c5f0ce338d73f4)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/soc/ti/wkup_m3_ipc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/ti/wkup_m3_ipc.c b/drivers/soc/ti/wkup_m3_ipc.c
index 498e1f7..0036525 100644
--- a/drivers/soc/ti/wkup_m3_ipc.c
+++ b/drivers/soc/ti/wkup_m3_ipc.c
@@ -122,7 +122,7 @@ static void wkup_m3_scale_data_fw_cb(const struct firmware *fw, void *context)
 	unsigned long val, aux_base;
 	struct wkup_m3_scale_data_header hdr;
 	struct wkup_m3_ipc *m3_ipc = context;
-	struct device *dev = (struct device *)context;
+	struct device *dev = m3_ipc->dev;
 
 	if (!fw) {
 		dev_err(dev, "Voltage scale fw name given but file missing.\n");
-- 
1.7.9.5

